spec/facility_spec.rb

```diff
@@ -23,4 +34,133 @@
       expect(@facility.services).to eq(['New Drivers License', 'Renew Drivers License', 'Vehicle Registration'])
     end
   end
-end
+
+  describe '#register vehicle' do
+    it 'may offer vehicle registration' do
+      expect(@facility_1.add_service('Vehicle Registration')).to eq(["Vehicle Registration"])
+      expect(@cruz.registration_date).to be nil
+      expect(@facility_1.registered_vehicles).to eq([])
+      expect(@facility_1.collected_fees).to eq 0
+
+      expect(@facility_1.register_vehicle(@cruz)).to eq([@cruz])
+      expect(@cruz.registration_date).to eq(Date.today)
+      expect(@cruz.plate_type).to eq(:regular)
+      expect(@facility_1.registered_vehicles).to eq([@cruz])
+      expect(@facility_1.collected_fees).to eq 100
+
+      expect(@facility_1.register_vehicle(@camaro)).to eq([@cruz, @camaro])
+      expect(@camaro.registration_date).to eq(Date.today)
+      expect(@camaro.plate_type).to eq(:antique)
+
+      expect(@facility_1.register_vehicle(@bolt)).to eq([@cruz, @camaro, @bolt])
+      expect(@bolt.registration_date).to eq (Date.today)
+      expect(@bolt.plate_type).to eq(:ev)
+
+      expect(@facility_1.registered_vehicles).to eq([@cruz, @camaro, @bolt])
+      expect(@facility_1.collected_fees).to eq 325
+    end
+
+    it 'is does not offer registration' do
+      expect(@facility_2.registered_vehicles).to eq([])
+      expect(@facility_2.services).to eq([])
+      expect(@facility_2.register_vehicle(@bolt)).to eq(nil)
+      expect(@facility_2.registered_vehicles).to eq([])
+      expect(@facility_2.collected_fees).to eq 0
+    end
+  end
+
+  # Written test
+
+  it 'can change licence data to true' do
+    expect(@registrant_1.license_data).to eq({:written=>false, :license=>false, :renewed=>false})
+    expect(@registrant_1.permit?).to be true
+
+    expect(@facility_1.administer_written_test(@registrant_1)).to be false
+    expect(@facility_1.add_service('Written Test')).to eq(['Written Test'])
```

Comment:
`@facility_1.add_service('Written Test')` is a command that updates the `.services` method.